From 67c74b32a8833f25f937b4254ff4e1220e3e6bb0 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 25 Nov 2020 08:25:16 +0200
Subject: [PATCH] [SW-0] spi image: validate ppboot.bin size

We cannot allow large foot print of ppboot as it will increase timing
for BTL to copy into SRAM and by that will japrodize PCIe link
establishment due to breaking 20mS time limit of warm reset
especially critical for Gaudi1
Today size is almost above 16KB and for Goya2 it won't fit FW SRAM

Change-Id: I113fa198585de4ca74ab4cc40115dfc5a893661a
---

diff --git a/board/habanalabs/common/post-image-spi.sh b/board/habanalabs/common/post-image-spi.sh
index 76e24a8..4f2f085 100755
--- a/board/habanalabs/common/post-image-spi.sh
+++ b/board/habanalabs/common/post-image-spi.sh
@@ -27,15 +27,24 @@
 
 function get_image_size {
 	local idx=$1
+	local sz=0
 
 	case $idx in
 	1) # Preboot
-		stat -Lc "%s" $PREBOOT_BINFILE
+		sz=`stat -Lc "%s" $PREBOOT_BINFILE`
+		if [ $sz -ge 4096 ]; then
+			echo "Preboot binary image is too large"
+			exit 1
+		fi
 		;;
 	2) # U-Boot
-		stat -Lc "%s" $UBOOT_BINFILE
+		sz=`stat -Lc "%s" $UBOOT_BINFILE`
 		;;
 	esac
+
+	if [ $sz -ne 0 ]; then
+		echo $sz
+	fi
 }
 
 function get_image_loadaddr {
